code/parse/sexp.cpp

```diff
@@ -19863,6 +19864,31 @@ void sexp_trigger_submodel_animation_new(int n)
 	animationStartFunc(direction, forced || instant, instant, pause);
 }
 
+void sexp_update_moveable_animation(int node)
+{
+	bool is_nan, is_nan_forever;
+
+	auto ship_entry = eval_ship(node);
+	if (!ship_entry || !ship_entry->shipp)
+		return;
+
+	node = CDR(node);
+
+	SCP_string name(CTEXT(node));
+
+	node = CDR(node);
+
+	//For now this only contains integers. It is very much feasible though that certain moveables might be updateable with strings and other non-numbers. For this, the C-side of the moveable code already supports other types
+	std::vector<linb::any> args;
+
+	while(node >= 0) {
+		args.emplace_back(eval_num(node, is_nan, is_nan_forever));
```

Comment:
Just be aware that in the unlikely event of a number being NAN or NAN_FOREVER, `eval_num` will return 0.  You may want to `continue` or display a warning if you find a NAN.